shelley/chain-and-ledger/executable-spec/src/Cardano/Ledger/Val.hs

```diff
@@ -1,72 +1,65 @@
 {-# LANGUAGE BangPatterns #-}
-{-# LANGUAGE ConstraintKinds #-}
-{-# LANGUAGE DeriveGeneric #-}
-{-# LANGUAGE DerivingVia #-}
-{-# LANGUAGE EmptyDataDecls #-}
-{-# LANGUAGE FlexibleInstances #-}
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
-{-# LANGUAGE LambdaCase #-}
-{-# LANGUAGE MultiParamTypeClasses #-}
-{-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE StandaloneDeriving #-}
-{-# LANGUAGE TypeFamilies #-}
 
 -- | This module defines a generalised notion of a "value" - that is, something
 -- with which we may quantify a transaction output.
---
--- This module is intended for qualified import:
--- > import qualified Cardano.Ledger.Val as Val
 module Cardano.Ledger.Val
   ( Val (..),
-    (<+>),
-    (<->),
-    (<×>),
-    invert,
+    scale,
     sumVal,
     scaledMinDeposit,
-
-    -- * Re-exports
-    (Data.PartialOrd.<=),
-    (Data.PartialOrd.>=),
-    (Data.PartialOrd.==),
-    (Data.PartialOrd./=),
-    (Data.PartialOrd.>),
-    (Data.PartialOrd.<),
-    Data.PartialOrd.compare,
+    addrHashLen,
+    uint,
+    assetIdLen,
+    mapV,
+    unionWithV,
+    pointWise,
+    insertWithV,
   )
 where
 
-import Cardano.Binary
-  ( FromCBOR (..),
-    ToCBOR (..),
+import Data.Map.Internal
+  ( Map (..),
+    balanceL,
+    balanceR,
+    link,
+    link2,
+    singleton,
+    splitLookup,
   )
-import Cardano.Prelude (NFData (), NoUnexpectedThunks (..))
-import Data.Group (Abelian)
-import Data.PartialOrd hiding ((==))
-import qualified Data.PartialOrd
+import qualified Data.Map.Strict as Map
 import Data.Typeable (Typeable)
 import Shelley.Spec.Ledger.Coin (Coin (..))
 
-class
-  ( Abelian t,
-    Eq t,
-    PartialOrd t,
```

Comment:
Do we no longer require a partial order on `Val`s? E.g. this is no longer part of the specification?

Reply:
We do require a partial order, but it is not the one people think of when they use the opertions (<), (>), (==).  Haskell users are used to thinking that for every two objects x and y.  Exactly one and of the three comparisons below is true.

x == y
x < y
c > y

The spec makes it clear that what we really want is a pointwise comparison of these operations.  forall a,b in dom x `union` dom y . a < b,  where we lift a to zero when a is not in dom x , and we lift b to zero when b is not in dom y.  Instead of supplying these misleading operations, we supply the  pointWise operation. Thus where we might want to write:    x < y      we write with better clarity   pointWise (<) x y